packages/crypto/src/math.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Compute the modular square root using the Tonelli-Shanks algorithm.
+ */
+export const modSqrt = (x: bigint, p: bigint): bigint => {
+  if (p <= BigInt(0)) {
+    throw new Error("p must be positive");
+  }
+  const base = x % p;
+
+  // Check if p % 4 == 3 (applies to NIST curves P-256, P-384, and P-521)
+  if (testBit(p, 0) && testBit(p, 1)) {
+    const q = (p + BigInt(1)) >> BigInt(2);
+    const squareRoot = modPow(base, q, p);
+    if ((squareRoot * squareRoot) % p !== base) {
+      throw new Error("could not find a modular square root");
+    }
+    return squareRoot;
+  }
+
+  // Other elliptic curve types not supported
+  throw new Error("unsupported modulus value");
+};
+
+/**
+ * Test if a specific bit is set.
+ */
+export const testBit = (n: bigint, i: number): boolean => {
+  const m = BigInt(1) << BigInt(i);
+  return (n & m) !== BigInt(0);
+};
+
+/**
+ * Generate a random Uint8Array of a specific length.
+ */
+export const randomBytes = (length: number): Uint8Array => {
+  const array = new Uint8Array(length);
+  // @ts-ignore
```

Comment:
what is this @ts-ignore for? I'm a bit surprised this is causing typescript to complain, looks pretty straightforward 🤔

Reply:
I suppose it's because the context is ambiguous (node vs. web)... i.e. we could add a `/// <reference lib="dom" />` to reference web or `import crypto from crypto`